src/bun.js/module_loader.zig

```diff
@@ -312,12 +312,18 @@ pub const RuntimeTranspilerStore = struct {
             const promise = this.promise.swap();
             const globalThis = this.globalThis;
             this.poll_ref.unref(vm);
-            var specifier = if (this.parse_error == null) this.resolved_source.specifier else bun.String.createUTF8(this.path.text);
+
             const referrer = bun.String.createUTF8(this.referrer);
             var log = this.log;
             this.log = logger.Log.init(bun.default_allocator);
             var resolved_source = this.resolved_source;
-            resolved_source.source_url = specifier.dupeRef();
+            const specifier = brk: {
+                if (this.parse_error == null) {
```

Comment:
is this intentionally the opposite from before? `if (this.parse_error == null) ...` was previously `this.resolved_source.specifier`

Reply:
ur right